### PR TITLE
mraa.c: Replaced string values with #defines

### PR DESCRIPTION
**Pull request by Hbrinj**
Signed-off-by: Houman Brinjcargorabi <hbrinjcar@gmail.com>

### CODE REVIEW

File: src/mraa.c [hunk 1]
```diff
@@ -1210,7 +1210,7 @@ mraa_init_io(const char* desc)
         return NULL;
     }
 
-    if (strncmp(type, "GPIO", 4) == 0) {
+    if (strncmp(type, GPIO_KEY, 4) == 0) {
```
Comment (alext-mkrs) [inline src/mraa.c]: I'd suggest replacing magic "4" with an automatically computed length, for all these comparisons.

Comment (tingleby) [inline src/mraa.c]: sizeof(GPIO_KEY) should be replace by the compiler with a value.Not sure how well this works.

**Comment by Hbrinj** [comment]
Done

**Comment by arfoll** [comment]
Merged thanks!

**Pull request closed without merging**